src/commands.rs

```diff
@@ -162,52 +171,34 @@ pub struct ReadFlashParams {
     pub max_inflight: u32,
 }
 
-#[derive(PartialEq, Copy, Clone, Debug)]
+#[derive(Debug, Clone, Copy, PartialEq, Eq)]
 #[repr(C, packed(1))]
-pub struct ReadFlash {
-    pub base: Base,
+pub struct ReadFlashCommand {
+    pub base: CommandBase,
     pub params: ReadFlashParams,
 }
 
-#[allow(unused)]
-#[repr(u8)]
-#[derive(PartialEq, Copy, Clone, Debug)]
-enum Direction {
-    In,
-    Out,
-}
-
-#[repr(u8)]
-#[derive(PartialEq, Copy, Clone)]
-enum Status {
-    Success,
-    Failure,
-}
-
-#[derive(PartialEq, Copy, Clone)]
+#[derive(Debug, Clone, Copy, PartialEq, Eq)]
 #[repr(C, packed(1))]
 pub struct Response<'a> {
-    direction: Direction,
-    command: Code,
-    size: u16,
-    value: u32,
-    status: Status,
-    error: Error,
+    pub direction: u8,
+    pub command: CommandCode,
+    pub size: u16,
+    pub value: u32,
+    pub status: u8,
+    pub error: u8,
     pub data: &'a [u8],
 }
 
-// Size of sesponse without data reference
-pub const RESPONSE_SIZE: usize = 10;
-
 impl<'a> Response<'a> {
-    pub fn new(cmd: Code) -> Self {
+    pub fn new(cmd: CommandCode) -> Self {
         Response {
-            direction: Direction::Out,
+            direction: 1,
```

Comment:
I wonder if the version with enums is not more readable then the version with an integer constant?

Reply:
You are right, thank you!